code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -46,6 +46,10 @@
 	for(var/obj/structure/closet/closet in get_turf(src))
 		if(closet != src && closet.anchored != 1)
 			return FALSE
+	for(var/mob/living/silicon/ai/ai in get_turf(src))

```

Comment:
I wonder if it's cheaper to do two `for` loops vs. doing something like
```
for(var/atom/A in get_turf(src))
    if(isAI(A))
        return FALSE
    if(istype(A, /obj/structure/closet) && !A.anchored)
        return FALSE
return TRUE
```
Anyone know?

Reply:
@farie82 might?

Reply:
i'd think it would be cheaper to do 1

Reply:
Can also use `locate(mob/living/silicon/ai) in get_turf(src)` to find one thing.

I'd say one loop will be faster since it will still get all the contents of the turf and filter them. Twice in the current case. 
The given suggestion will get them once then loop through them. Which should be better performing in theory. But I'm not to sure how byond does this all
Can prob even do `var/A` instead of `var/atom/A`.

Reply:
Which one would be best recommended?
What I currently have, the suggestion made by @Citinited , or the usage of the `locate` proc?

Reply:
Why don't you put the change at line 199, inside the `close` loop instead

Reply:
So it would be something like this, @Crazylemon64 ?

```
for(var/obj/structure/closet/closet in get_turf(src))
	if(closet != src && closet.anchored != 1)
		return FALSE
	for(var/mob/living/silicon/ai/ai in get_turf(src))
                if(ai)
	               return FALSE
```
Because if so, what @Citinited suggested would look more cleaner.

Reply:
The second for loop shouldn't be inside the first, and you don't need the `if(ai)` bit because that will always be TRUE (just `return FALSE`)

Reply:
It'd be in that `for(mob...)` loop - alongside that mulebot check